### DISCUSSION TITLE
HostResourceProvider resource attributes, hostname wrong

### DISCUSSION THREAD

**Discussion by krishnaexplore** [open-telemetry/opentelemetry-java-instrumentation#10938] [Q&A]
I'm using opentelemetry java agent version 1.31.0

seems HostResourceProvider not pickup right hostname in amazon linux, I got below values, even though `/etc/hostname` has right name,

```
 "host.arch": "amd64",     
 "host.name": "32bf17e5674f",

```

 what could be the problem here

**Comment by laurit** [comment]
What does the `hostname` command return? Agent gets the host name with https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/080f8577e526617c80f834f070d32b434026482d/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/HostResource.java#L29 Cloud vendor specific resource providers may overwrite this, are you using aws resource providers?

**Comment by krishnaexplore** [comment]
oh got the problem, my app running inside container.. seems i need to update /etc/hosts of docker container
